web/frontend/src/components/slam.vue

```diff
@@ -1,193 +1,208 @@
-   * await mapPromise;
-   * await posePromise;
-   */
-  await fetchSLAMImageMap(name);
+const scheduleRefresh2d = (name: string, time: string) => {
+  const timeoutCallback = async () => {
+    try {
+      const results = await refresh2d(name);
+      if (results && results.length === 2) {
+        [pointcloud, pose] = results;
+        pointCloudUpdateCount += 1;
+      }
+    } catch (error) {
+      displayError(error);
+      return;
+    }
+    if (refresh2DCancelled) {
+      return;
+    }
+    scheduleRefresh2d(name, time);
+  };
+  slam2dTimeoutId = window.setTimeout(timeoutCallback, Number.parseFloat(time) * 1000);
+};
+
+const scheduleRefresh3d = (name: string, time: string) => {
+  const timeoutCallback = async () => {
+    try {
+      const result = await fetchSLAMMap(name);
+      if (result) {
+        pointcloud = result;
+        pointCloudUpdateCount += 1;
+      }
+    } catch (error) {
+      displayError(error);
+      return;
+    }
+    if (refresh3DCancelled) {
+      return;
+    }
+    scheduleRefresh3d(name, time);
+  };
+  slam3dTimeoutId = window.setTimeout(timeoutCallback, Number.parseFloat(time) * 1000);
 };
 
-// eslint-disable-next-line require-await
 const updateSLAM2dRefreshFrequency = async (name: string, time: 'manual' | 'off' | string) => {
-  window.clearInterval(slam2dIntervalId);
+  refresh2DCancelled = true;
+  window.clearTimeout(slam2dTimeoutId);
 
   if (time === 'manual') {
-    refresh2d(name);
-
+    try {
+      const results = await refresh2d(name);
+      if (results && results.length === 2) {
+        [pointcloud, pose] = results;
+        pointCloudUpdateCount += 1;
+      }
+    } catch (error) {
+      displayError(error);
+    }
   } else if (time === 'off') {
     // do nothing
   } else {
-
-    refresh2d(name);
-    slam2dIntervalId = window.setInterval(() => {
-      refresh2d(name);
-    }, Number.parseFloat(time) * 1000);
+    refresh2DCancelled = false;
+    scheduleRefresh2d(name, time);
   }
 };
 
-/*
- * const updateSLAM3dRefreshFrequency = (name: string, time: 'manual' | 'off' | string) => {
- *   clearInterval(slam3dIntervalId);
- *   if (time === 'manual') {
- *     fetchSLAMMap(name);
- *   } else if (time === 'off') {
- *     // do nothing
- *   } else {
- *     fetchSLAMMap(name);
- *     slam3dIntervalId = window.setInterval(() => {
- *       fetchSLAMMap(name);
- *     }, Number.parseFloat(time) * 1000);
- *   }
- * };
- */
+const updateSLAM3dRefreshFrequency = async (name: string, time: 'manual' | 'off' | string) => {
+  refresh3DCancelled = true;
+  window.clearTimeout(slam3dTimeoutId);
+
+  if (time === 'manual') {
+    try {
+      const result = await fetchSLAMMap(name);
+      if (result) {
+        pointcloud = result;
+        pointCloudUpdateCount += 1;
+      }
+    } catch (error) {
+      displayError(error);
+    }
+  } else if (time === 'off') {
+    // do nothing
+  } else {
+    refresh3DCancelled = false;
+    scheduleRefresh3d(name, time);
```

Comment:
This now uses  setTimeout rather than setInterval to ensure that we only have one SLAM request to the backend at a given time to reduce un necessary network calls.